docs/docs/tutorial/creating-a-source-plugin/part-3/index.mdx

```diff
@@ -0,0 +1,587 @@
+
+</Announcement>
+
+In this tutorial you'll use the `@link` extension to create a foreign-key relationship between your `Post` and `Author` nodes. This relationship can either already exist in your data source (and you want to replicate it) or it can be a completely new association you want to create.
+
+You can also use the [`@proxy` directive](/docs/reference/graphql-data-layer/schema-customization/#aliasing-fields) to model your GraphQL types to your liking but it won't be covered in this tutorial.
+
+<Announcement>
+
+**Foreign-key relationship:** This term comes from relational databases. A foreign-key is a set of attributes in a table that refers to the primary key of another table. The foreign-key links these two tables.
+
+</Announcement>
+
+When inspecting the incoming `Post` and `Author` data, do you see the foreign-key between both data sets?
+
+`Post`:
+
+```json
+{
+  id: '1',
+  slug: 'post-1',
+  title: 'The first post',
+  image: {
+    url: 'https://images.unsplash.com/photo-1615751072497-5f5169febe17?fm=jpg',
+    alt: 'brown and white long coated dog',
+    width: 3024,
+    height: 4032
+  },
+  author: 'Jay Gatsby'
+}
+```
+
+`Author`:
+
+```json
+{ 
+  id: '1',
+  name: 'Jay Gatsby'
+}
+```
+
+The `author` field on `Post` is the foreign-key to the `Author` data set. It is the unique identifier that could link both data sets together.
+
+<Announcement>
+
+**Key Gatsby Concept** 💡
+
+The goal of a foreign-key relationship through `@link` in Gatsby is to expand the provided field value to a full node. You can enrich the information of a node by adding relevant/additional data from another node.
+
+This is a really powerful ability in Gatsby's data layer.
+
+Imagine this example: "Team Gatsby" likes to author their blog posts in Markdown, "Team Daisy" is responsible for keeping author information up do date and keeps track of that in a CMS. As long as both teams use the same unique identifier, both nodes can be linked together in Gatsby's data layer. Both teams can work independently on their data.
+
+</Announcement>
+
+In the previous section you already defined the GraphQL types for `Post` and `Author` so all the necessary boilerplate is already done. You can now use the `@link` extension (as a directive in the SDL). Before showing that, here's a short explanation on the `@link` extension:
+
+- `@link` can receive two arguments (`by` and `from`) but if no argument is given, Gatsby will use the `id` field as the foreign-key field (equivalent to `@link(by: "id")`). You'll use this default behavior in [Part 6](/docs/tutorial/creating-a-source-plugin/part-6/) for the Image CDN feature.
+- `by` is the field you link **to** (on the target node).
+- `from` is the field you link **from** (on the current node).
```

Comment:
There is usage of `from` here, so I wonder it warrants a note saying that you would want to use that argument if you want different name for field in graphql that what Node has named the value?

So as example if node has `author_id` field, you wouldn't want field like that in schema. Instead you want `author` field that uses current's node `author_id` for source of value to match target node on?

Feel free to ignore (or maybe I'll discard it if there is mention of it later on)